### ISSUE TITLE
all-positions

### ISSUE THREAD

**Issue by arthurhjorth** [NetLogo/NetLogo#1042]
`position` currently only reports the position of the first found, similar element in a list. It would be great to have an `all-positions` reporter that gives you a list of all of the positions of the argument in a provided list. Wouldn't it?

**Comment by SethTisue** [comment]
I doubt this comes up often enough to be worth its weight in the language.

**Comment by arthurhjorth** [comment]
Yeah, that seemed to be Bryan's take on it too. I remember wanting it quite often a few years ago and writing clunky implementations, but that could have been because I wasn't got enough with NetLogo list and string primitives to do a better/prettier job. I don't remember the exact usecase any longer, but that's certainly possible.

**Comment by qiemem** [comment]
For reference, implementation in NetLogo:

```
to-report all-positions [ x lst ]
  report map first filter [ last ? = x ] n-values (length lst) [ list ? item ? lst ]
end
```

Some context: this came up in class yesterday (specifically, someone wanted to find all the indices of the largest value). Arthur and I stuttered for a few minutes before essentially saying "throw `n-values`, `filter` and `map` in there in some order...".

**Comment by nicolaspayette** [comment]
My 2¢: a (bundled?) `list` extension would be a great place to put all these occasionally useful list primitives... (see also #931).

**Comment by ERSUCC** [comment]
This is now implemented in the Functional Programming extension as [fp:find-indices](https://github.com/NetLogo/FP-Extension?tab=readme-ov-file#fpfind-indices).
